config/crd/bases/agent-install.openshift.io_agents.yaml

```diff
@@ -168,6 +168,18 @@ spec:
                       the Agent
                     type: string
                 type: object
+              hostname:
+                description: "Hostname is the name that will be assigned to the host
+                  when the cluster is eventually installed. This will be different
+                  to the host name included in the inventory if that is a forbidden
+                  host name (like localhost, localhost4 or localhost6) or if the user
+                  has explicitly requested a different name using the .spec.hostname
+                  field. \n When the host name included in the inventory is forbidden
+                  and the user hasn't explicitly requested a different one, this will
+                  be calculated from the MAC address of the host. For example, if
+                  the MAC address of the host is A8:CD:16:AE:79:01 then this will
```

Comment:
There's no "The MAC address" for a host, what if it has multiple interfaces? Phrasing should be adjusted

Reply:
@omertuc I wanted to be vague there on purpose, because the algorithm is not deterministic, and I think explaining it to the user will not be very useful. It would be something like this:

>  The host name will be calculated from the MAC address of one of the network interfaces
>  of the host that has both a MAC address and a non-local IP address.

Note the "one of the network interfaces" part. That is what the code does; it iterates the list of network interfaces
of the host and picks the first that has a MAC and a non-local IP address. If I understand correctly there is no
guarantee of any kind of ordering in that list, so the selected network interface may change if the agent or
the service decide to sort this list in a different way. @mkowalski mentioned that we could make this selection
deterministic by looking at the cluster networks and selecting a network interface that is connected to the
machine network of the cluster. I thought that the benefit of doing that is not worth the effort: all we need here
is a unique host name. Do we want make this deterministic?

Reply:
I have no opinion about the interface selection itself, just about the description being literally incorrect. 

Vagueness can be achieved by just saying a instead of the.

Ideally we should pick the machine network interface, I think we already have functions that do that if you feel like reusing them, but again I'm unopinionated about it

Reply:
The algorithm should definitely be deterministic